l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -0,0 +1,650 @@
+            require(msg.value == _amount, "L1SharedBridge: msg.value not equal to amount");
+        } else {
+            // The Bridgehub also checks this, but we want to be sure
+            require(msg.value == 0, "ShB m.v > 0 b d.it");
+
+            uint256 amount = _depositFunds(_prevMsgSender, _l1Token, _amount); // note if _prevMsgSender is this contract, this will return 0. This does not happen.
+            require(amount == _amount, "3T"); // The token has non-standard transfer logic
+        }
+
+        if (!hyperbridgingEnabled[_chainId]) {
+            chainBalance[_chainId][_l1Token] += _amount;
+        }
+        // Note we don't save the deposited amount, as this is for the base token, which gets sent to the refundRecipient if the tx fails
+    }
+
+    /// @dev Transfers tokens from the depositor address to the smart contract address
+    /// @return The difference between the contract balance before and after the transferring of funds
+    function _depositFunds(address _from, address _token, uint256 _amount) internal returns (uint256) {
+        IERC20 token = IERC20(_token);
+        uint256 balanceBefore = token.balanceOf(address(this));
+        token.safeTransferFrom(_from, address(this), _amount);
+        uint256 balanceAfter = token.balanceOf(address(this));
+
+        return balanceAfter - balanceBefore;
+    }
+
+    /// @notice used by requestL2TransactionTwoBridges in Bridgehub
+    /// specifies called chainId and caller, and requested transaction in _data.
+    /// currently we only support a single tx, depositing.
+    function bridgehubDeposit(
+        uint256 _chainId,
+        address _prevMsgSender,
+        uint256, // l2Value, needed for Weth deposits in the future
+        bytes calldata _data
+    ) external payable override onlyBridgehub returns (L2TransactionRequestTwoBridgesInner memory request) {
+        require(l2BridgeAddress[_chainId] != address(0), "ShB l2 bridge not deployed");
+
+        (address _l1Token, uint256 _depositAmount, address _l2Receiver) = abi.decode(
+            _data,
+            (address, uint256, address)
+        );
+        require(_l1Token != l1WethAddress, "ShB: WETH deposit not supported");
+        require(bridgehub.baseToken(_chainId) != _l1Token, "ShB: baseToken deposit not supported");
+
+        uint256 amount;
+        if (_l1Token == ETH_TOKEN_ADDRESS) {
+            amount = msg.value;
+            require(_depositAmount == 0, "ShB wrong withdraw amount");
+        } else {
+            require(msg.value == 0, "ShB m.v > 0 for BH d.it 2");
+            amount = _depositAmount;
+
+            /// This breaks the _depositeFunds function, it returns 0, as we are withdrawing funds from ourselves, so our balance doesn't increase
+            /// This should not happen, this bridge only calls the Bridgehub if Eth is the baseToken or for wrapped base token deposits
+            require(_prevMsgSender != address(this), "ShB calling itself");
```

Comment:
Is this needed or it can be removed as the check from `bridgehubDepositBaseToken`?

Reply:
I thought I removed this.